tasm-lib/src/arithmetic/u128/lt_u128.rs

```diff
@@ -0,0 +1,231 @@
+                .pad_using(4, |_| 0)
+                .collect_vec();
+
+            [
+                init_stack,
+                rhs_limbs.into_iter().map(|l| bfe!(l)).rev().collect_vec(),
+                lhs_limbs.into_iter().map(|l| bfe!(l)).rev().collect_vec(),
+            ]
+            .concat()
+        }
+    }
+
+    impl Closure for LtU128 {
+        fn rust_shadow(&self, stack: &mut Vec<triton_vm::prelude::BFieldElement>) {
+            let mut left_limbs = vec![];
+            for _ in 0..4 {
+                left_limbs.push(stack.pop().unwrap().value() as u32);
+            }
+            let mut right_limbs = vec![];
+            for _ in 0..4 {
+                right_limbs.push(stack.pop().unwrap().value() as u32);
+            }
+
+            let right = right_limbs
+                .into_iter()
+                .rev()
+                .fold(0u128, |acc, limb| (acc << 32) | (limb as u128));
+            let left = left_limbs
+                .into_iter()
+                .rev()
+                .fold(0u128, |acc, limb| (acc << 32) | (limb as u128));
+
+            stack.push(bfe!((left < right) as u64));
+        }
+
+        fn pseudorandom_initial_state(
+            &self,
+            seed: [u8; 32],
+            _bench_case: Option<crate::snippet_bencher::BenchmarkCase>,
+        ) -> Vec<triton_vm::prelude::BFieldElement> {
+            let mut rng: StdRng = SeedableRng::from_seed(seed);
+            self.prepare_stack(rng.gen(), rng.gen())
+        }
+    }
+
+    #[test]
+    fn lt_u128_standard_test() {
+        ShadowedClosure::new(LtU128).test()
+    }
+
+    fn test_rust_tasm_equivalence(lhs: u128, rhs: u128) {
+        let initial_state = InitVmState::with_stack(LtU128.prepare_stack(lhs, rhs));
+        test_rust_equivalence_given_execution_state(&ShadowedClosure::new(LtU128), initial_state);
+    }
+
+    #[test]
+    fn lt_u128_unit_test() {
+        test_rust_tasm_equivalence(18446744073709551616, 4294967296);
+        test_rust_tasm_equivalence(4294967296, 18446744073709551616);
+    }
+
+    #[test]
+    fn lt_u128_edge_cases_test() {
+        let boundary_points = [0u128, 1 << 32, 1 << 64, 1 << 96, u128::MAX];
+        for left_boundary in boundary_points {
+            for lhs in [
+                left_boundary.checked_sub(1),
+                Some(left_boundary),
+                left_boundary.checked_add(1),
+            ]
+            .into_iter()
+            .flatten()
+            {
+                for right_boundary in boundary_points {
+                    for rhs in [
+                        right_boundary.checked_sub(1),
+                        Some(right_boundary),
+                        right_boundary.checked_add(1),
+                    ]
+                    .into_iter()
+                    .flatten()
+                    {
+                        test_rust_tasm_equivalence(lhs, rhs);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
I like the flattening trick.
Here's another neat trick: `Itertools` provides a cartesian product of iterators.

```rust
#[test]
fn lt_u128_edge_cases_test() {
    let boundary_points = [0, 1 << 32, 1 << 64, 1 << 96, u128::MAX]
        .into_iter()
        .flat_map(|point| [point.checked_sub(1), Some(point), point.checked_add(1)])
        .flatten()
        .collect_vec();

    for (&lhs, &rhs) in boundary_points.iter().cartesian_product(&boundary_points) {
        test_rust_tasm_equivalence(lhs, rhs);
    }
}
```

I'm gonna stop now. :blush: